Build failure with custom LuaJIT (OpenResty fork)

I'm trying to build luvi with a custom LuaJIT, which is the latest release build from the OpenResty fork. This produces the following build error:

```
luajit: unknown luaJIT command or jit.* modules not installed
luajitluajit: unknown luaJIT command or jit.* modules not installed
: unknown luaJIT command or jit.* modules not installed
make[2]: *** [jitted_tmp/src/lua/luvibundle.lua_luvi_generated.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: *** [jitted_tmp/deps/lpeg/re.lua_luvi_generated.o] Error 1
make[2]: *** [jitted_tmp/src/lua/init.lua_luvi_generated.o] Error 1
luajit: unknown luaJIT command or jit.* modules not installed
make[2]: *** [jitted_tmp/src/lua/luvipath.lua_luvi_generated.o] Error 1
make[1]: *** [CMakeFiles/luvi.dir/all] Error 2
make: *** [all] Error 2
```

LuaJIT was built using the standard `make amalg`, so I'm not sure why I'm getting this error. Setting `LUA_PATH` to LuaJIT's `jit` directory as explained in https://github.com/luvit/luvi/commit/a9dc8dae2eb69d4a754a0cf5c08622d70f6b7067 didn't help. Assistance would be appreciated.

Complete build logs: https://gist.github.com/carlocab/2a6c25344be31127f61635d8978b3720

These were generated while attempting a build in Homebrew, but I can reproduce this without Homebrew using just a `make tiny` with the appropriate environment variables for set to enable building with a shared libluv. (I'm actually doing this to assist with Homebrew packaging of luvit. The latest luvi release doesn't build on Apple Silicon.)

I should also actually mention that using a shared libluv also causes errors with how `LPEGLIB_DIR` is set. Without manual intervention, I get this build error:
```
CMake Error in CMakeLists.txt:
  Cannot find source file:

    /tmp/luvit--luvi-20210121-94985-vucl7x/tmp/luvit--luvi-20210121-94985-vucl7x/deps/lpeg/re.lua
```
The correct path to `re.lua` is `tmp/luvit--luvi-20210121-94985-vucl7x/deps/lpeg/re.lua`. Not sure why the prefix is getting doubled there. Setting
```
export LPEGLIB_DIR=deps/lpeg
```
fixes the problem.

--- a/cmake/Modules/LuaJITAddExecutable.cmake
+++ b/cmake/Modules/LuaJITAddExecutable.cmake
@@ -26,15 +26,28 @@ MACRO(LUAJIT_add_custom_commands luajit_target)
 
       set(generated_file "${CMAKE_BINARY_DIR}/jitted_tmp/${stripped_file}_${luajit_target}_generated${CMAKE_C_OUTPUT_EXTENSION}")
 
-      add_custom_command(
-        OUTPUT ${generated_file}
-        MAIN_DEPENDENCY ${source_file}
-        COMMAND "LUA_PATH=${LUA_PATH}" luajit
-        ARGS -b ${LJ_BYTECODE_OPTS}
-          ${source_file}
-          ${generated_file}
-        COMMENT "Building Luajitted ${source_file}: ${generated_file}"
-      )
+      # Only use LUA_PATH if it's set
+      IF(LUA_PATH)
+        add_custom_command(
+          OUTPUT ${generated_file}
+          MAIN_DEPENDENCY ${source_file}
+          COMMAND "LUA_PATH=${LUA_PATH}" luajit
+          ARGS -b ${LJ_BYTECODE_OPTS}
+            ${source_file}
+            ${generated_file}
+          COMMENT "Building Luajitted ${source_file}: ${generated_file}"
+        )
+      ELSE()
+        add_custom_command(
+          OUTPUT ${generated_file}
+          MAIN_DEPENDENCY ${source_file}
+          COMMAND luajit
+          ARGS -b ${LJ_BYTECODE_OPTS}
+            ${source_file}
+            ${generated_file}
+          COMMENT "Building Luajitted ${source_file}: ${generated_file}"
+        )
+      ENDIF()
 
       get_filename_component(basedir ${generated_file} PATH)
       file(MAKE_DIRECTORY ${basedir})

